src/Fixer/PhpUnit/PhpUnitTestClassRequiresCoversFixer.php

```diff
@@ -63,14 +63,21 @@ public function isCandidate(Tokens $tokens)
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $phpUnitIndicator = new PhpUnitIndicator();
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
 
         for ($index = $tokens->count() - 1; $index >= 0; --$index) {
             if (!$tokens[$index]->isGivenKind(T_CLASS)) {
                 continue;
             }
 
-            if (!$phpUnitIndicator->isPhpUnitClass($tokens, $index)) {
+            $prevIndex = $tokens->getPrevMeaningfulToken($index);
+
+            // don't add `@covers` annotation for abstract base classes
+            if ($tokens[$prevIndex]->isGivenKind(T_ABSTRACT)) {
+                return false;
```

Comment:
this statement looks odd to me, as `applyFix` is not suppose to return a value I think?

Reply:
great catch, fixed